### PR TITLE
* Use native http_build_query() to build query string (PHP 5, PHP 7, PHP 8)

### PR DESCRIPTION
**Pull request by timint**
No need to invent the wheel 😁

### CODE REVIEW

File: src/HostedService/HostedRequest.php [hunk 1]
```diff
@@ -45,12 +45,7 @@ function __construct($config)
     public function doRequest()
     {
         $fields = $this->prepareRequest();
-
-        $fieldsString = "";
-        foreach ($fields as $key => $value) {
-            $fieldsString .= $key . '=' . $value . '&';
-        }
-        rtrim($fieldsString, '&');
+        $fieldsString = http_build_query($fields, '', '&');
```
Comment (alexanderwiden95) [inline src/HostedService/HostedRequest.php:48]:
```suggestion
        $fieldsString = urldecode(http_build_query($fields));
```

**Review by alexanderwiden95 — CHANGES_REQUESTED** [review]

**Comment by timint** [comment]
You can't urldecode a complete url. urldecode is for parameter values. Why would you need decode?

**Comment by alexanderwiden95** [comment]
`$fieldsString` is the arguments passed in `CURLOPT_POSTFIELDS`, not the complete URL

**Comment by timint** [comment]
My misstake, yes it's the post params only.
I mean why would you urldecode them? This is a serialized string.
What would you do if any of the values holds the characters [ ] = ?

Invalid: foo=abc&123
Valid: foo=abc%26123

**Comment by alexanderwiden95** [comment]
The problem is that "message" can contain signs such as "%2B" ("+"-sign) which will result in a incorrect formatted string if encoded.
In my testing the non-decoded `$fieldsString` results in a "311 (BAD_MAC)" since the "%2B" now isn't representing a plus sign.

You're correct in the fact that other characters such as []=?, and so on could make the call invalid.

**Comment by timint** [comment]
That is alarming. The original code lines neither do urlencode. It seems like if Svea Ekonomi are not decoding the data on their end. Which requires the API user to send the query without encoding.

This should be escalated back to the API developers.  Will you do it?

**Pull request closed without merging**